### ISSUE TITLE
Bug: Specificity problem with padding / margin and "constituent" styles like padding-left

### ISSUE THREAD

**Issue by earshinov** [Stranger6667/css-inline#148]
An example:

```python
import css_inline

html = """<html>
<head>
    <title>Test</title>
    <style>
        .test { padding-left: 16px; }
        h1 { padding: 0; }
    </style>
</head>
<body>
    <h1 class="test">Should have padding: 0 0 0 16px;</h1>
</body>
</html>"""

print(css_inline.inline(html))

#<html><head>
#    <title>Test</title>
#    <style>
#        .test { padding-left: 16px; }
#        h1 { padding: 0; }
#    </style>
#</head>
#<body>
#    <h1 class="test" style="padding-left: 16px;padding: 0;">Should have padding: 0 0 0 16px;</h1>
#
#</body></html>
```

As specificity rules state, the styles bound to `.test` selector should have priority over those bound to `h1`. Generally css-inline handles specificity well, but in the example given above, styles inlining should result in `padding: 0;padding-left: 16px;` (equivalent to `padding: 0 0 0 16px;`) instead of `padding-left: 16px;padding:0;` (equivalent to `padding: 0;`).

Expected result:

```html
<h1 class="test" style="padding: 0;padding-left: 16px;">
```

❌ Actual result: 

```html
<h1 class="test" style="padding-left: 16px;padding: 0;">
```

Version information:

```
$ pip show css_inline | grep Version
Version: 0.8.2
```

**Comment by earshinov** [comment]
I think, to resolve this you would need to collect _all_ selectors first, then sort them by (specificity, order) and only then apply them to matching elements… Code pointer: https://github.com/Stranger6667/css-inline/blob/cd70af194f36eb26808ee1b385ef266771ff172a/css-inline/src/lib.rs#L369

**Comment by Stranger6667** [comment]
Thank you for reporting and sorry for the delay! Should be fixed by #159

**Comment by Stranger6667** [comment]
@earshinov the fix is released in `0.8.4`
